lib/MusicBrainz/Server/Edit/Historic/Relationship.pm

```diff
@@ -212,30 +212,40 @@ sub _display_relationships {
     my ($self, $data, $loaded) = @_;
     return [
         map {
+            my $entity0_type = $_->{entity0_type};
+            my $entity1_type = $_->{entity1_type};
             my $model0 = type_to_model( $_->{entity0_type} );
             my $model1 = type_to_model( $_->{entity1_type} );
             my $entity0_id = $_->{entity0_id};
             my $entity1_id = $_->{entity1_id};
             my $entity0 = $loaded->{ $model0 }{ $entity0_id } ||
-                $self->c->model($model0)->_entity_class->new( name => $_->{entity0_name});
+                $self->c->model($model0)->_entity_class->new(
+                    entity_type => $entity0_type,
```

Comment:
Is `entity_type` needed there? I thought the entity class itself would already have a `sub entity_type { 'foo' }` implemented.